biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -2708,22 +2779,46 @@ subroutine terminate_patches(currentSite)
 
     currentPatch => currentSite%youngest_patch
     do while(associated(currentPatch)) 
-       
-       if(currentPatch%area <= min_patch_area)then
+       lessthan_min_patcharea_if: if(currentPatch%area <= min_patch_area)then
           
-         
+          nocomp_if: if (hlm_use_nocomp .eq. itrue) then
+
+             gotfused = .false.
+             patchpointer => currentSite%youngest_patch
+             do while(associated(patchpointer))
+                if ( .not.associated(currentPatch,patchpointer) .and. &
+                     patchpointer%nocomp_pft_label .eq. currentPatch%nocomp_pft_label .and. &
+                     patchpointer%anthro_disturbance_label .eq. currentPatch%anthro_disturbance_label .and. &
+                     .not. gotfused) then
+
+                   call fuse_2_patches(currentSite, patchpointer, currentPatch)
+                   
+                   gotfused = .true.
+                else
+                   patchpointer => patchpointer%older
+                endif
+             end do
+
+             if ( .not. gotfused ) then
+                !! somehow didn't find a patch to fuse with.
+                write(fates_log(),*) 'Warning. small nocomp patch wasnt able to find another patch to fuse with.', &
```

Comment:
this is I think a separate point from the number of patches that can be allocated.  In order to have such a small patch, it would have needed to have been carved off from another patch.  But then somehow couldn't have gotten fused back into it.  So the kind of edge case that a warning statement would be useful to know if it occurred.